### PR TITLE
Remove meshconfig/injector configmap mount

### PR DESCRIPTION
**Pull request by howardjohn**
We have had code to read these from configmap for a while. This makes us use that. The benefit is that the configmap watch updates immediately instead of 1-2 minutes later, and works well in external istiod.

As part of this, update galley code to also read from the s.environment mesh config. In practice, this mesh config is never actually used. In theory it feeds analyzers, but only one uses it, and that one doesn't run in cluster.

### CODE REVIEW

**Review by stevenctl — APPROVED** [review]

**Review by costinm — COMMENTED** [review]
To make sure I understand, Istiod will load MeshConfig from CR in all cases ( or all cases where K8S is enabled ) ? But still be able to load from file somehow ? 

More comments would help.

**Comment by howardjohn** [comment]
current logic is to load from file if it exists, otherwise load from configmap. This is in https://github.com/istio/istio/blob/18b3d35de0523862a389553b50b13ee4e73f4530/pilot/pkg/bootstrap/mesh.go#L38.

I didn't change any logic there, just removing the mounts so that we hit the "load from configmap" code. We handle case of no k8s, so we won't just fail there

File: pilot/pkg/bootstrap/configcontroller.go [hunk 1]
```diff
@@ -241,8 +242,13 @@ func (s *Server) initInprocessAnalysisController(args *PilotArgs) error {
 	processingArgs := settings.DefaultArgs()
 	processingArgs.KubeConfig = args.RegistryOptions.KubeConfig
 	processingArgs.WatchedNamespaces = args.RegistryOptions.KubeOptions.WatchedNamespaces
-	processingArgs.MeshConfigFile = args.MeshConfigFile
 	processingArgs.EnableConfigAnalysis = true
+	meshSource := mesh.NewInmemoryMeshCfg()
```
Comment (irisdingbj) [inline pilot/pkg/bootstrap/configcontroller.go:246]: Does this mean the InprocessAnalysisController will always use the defaultMeshconfig? How about the cases that user defines the meshConfig via file or ConfigMap?

**Review by hzxuzhonghu — APPROVED** [review]

File: pilot/pkg/bootstrap/configcontroller.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/bootstrap/configcontroller.go:246]: No, L247-250 update it when it changes

Comment (irisdingbj) [reply pilot/pkg/bootstrap/configcontroller.go:246]: ah, got it

**Review by irisdingbj — APPROVED** [review]

**Comment by ericvn** [comment]
It seems this PR has caused the https://preliminary.istio.io/latest/docs/setup/additional-setup/external-controlplane/ test to start failing.

**Pull request merged**